db_stress_tool/db_stress_test_base.cc

```diff
@@ -1563,6 +1563,8 @@ Status StressTest::TestCheckpoint(ThreadState* thread,
   if (!s.ok()) {
     fprintf(stderr, "A checkpoint operation failed with: %s\n",
             s.ToString().c_str());
+    delete checkpoint;
+    checkpoint = nullptr;
```

Comment:
Why not just move the delete checkpoint lines @1511 outside of the s.ok() block, either at the "top-level" or inside the previous s.ok() block (in which the checkpoint was successfully created)?

Reply:
Will do, thanks!